src/terra-dev-site/guide/DockerDevelopmentEnvironment.i.guide.md

```diff
@@ -0,0 +1,91 @@
+
+After you've finished running tests, run:
+```bash
+docker-compose down
+```
+This will stop the standalone-chrome service that will otherwise be left running on your device.
+
+
+### Terminal access
+The docker development environment comes pre-loaded with oh-my-zsh and bash. There are two methods of running the terminal on the docker container. Start the docker container to a terminal session through docker-compose or use docker to attach to a running docker image.
+
+To launch to the terminal on the dev image run:
+
+```
+docker-compose run --service-ports dev /bin/zsh
+```
+To attach to a running docker container run:
+
+```
+docker exec -it <hash for running docker container> /bin/zsh
+```
+
+### Local dependencies
+Local dependencies are supported in the docker development environment but require an additional step. For the docker container to be able to find the local dependency, it must be volume mounted to the docker container.
+
+For example:
+
+package.json
+```json
+"dependencies": {
+    "classnames": "file:../classnames",
+}
+```
+
+docker-compose.yml
+```yml
+dev:
+  image: "terra-ui:dev"
+  build:
+    context: ./
+    args:
+      - npm_install=false
+  ports:
+    - 8080:8080
+  volumes:
+    - .:/opt/module
+    - ../classnames:/opt/classnames
+```
+
+### Sharing node_modules
+The docker development environment intentionally will write out node_modules to the shared host volume. This provides several benefits:
+* Simpler volume mounting, everything edited in the repo is shared to the docker container.
+* Node module cacheing. For the development containers npm install is run at run time of the docker container (opposed to build time). This is done to allow npm install to discover local dependencies. To avoid long startup times node_modules are cached to the local volume. If node_modules are found npm install is not run at run time.
+
+### Troubleshooting
+#### Error: Missing binding
+This error means that a node_module install for one environment is being run in another. Likely this means that you ran npm install on your host OS and are now trying to use that module in your docker container or the opposite. To resolve the issue delete the node_modules folder and run npm_install where you are attempting to start the server.
+
+#### Dependency changes not taking effect.
+npm install is run at run time of the docker container. If node_modules are found npm install is not run at run time. If dependencies have changed the script checking for node_modules isn't smart enough to re-run npm install. To resolve the issue either open a terminal docker container and run npm install or delete the node_modules folder.
```

Comment:
npm install is ~run~ *triggered* a run time of the docker container. If **a local node_modules directory** is found, npm install ~is not run at run time~ **will not run**. 

To resolve,
1. Start a new docker container via `docker-compose run` and run `npm install` (should it be `npm run clean:install`?)
2. delete the locale `node_modules` folder and start a new container

Reply:
These are either/or steps, you can either resolve through the terminal or delete node_modules and run the server. You wouldn't do both.